src/home/homework-quota.c

```diff
@@ -52,7 +53,10 @@ int home_update_quota_classic(UserRecord *h, const char *path) {
         if (r < 0)
                 return log_error_errno(r, "Failed to determine block device of %s: %m", path);
         if (devno == 0)
-                return log_error_errno(SYNTHETIC_ERRNO(ENODEV), "File system %s not backed by a block device.", path);
+                return log_full_errno(
+                                detect_container() > 0 ? LOG_DEBUG : LOG_WARNING,
+                                SYNTHETIC_ERRNO(ENODEV),
+                                "File system %s not backed by a block device.", path);
```

Comment:
hmm, i think we should go further here. should not be a warning at all. i.e. maybe rework the whole thing so that we return 0 here if quota is not applicable and 1 if it is... And then never generate beyond LOG_INFO (probably just LOG_DEBUG) for the cases where quota is not available, because that is not an error in many cases.